app/app/utils.py

```diff
@@ -458,6 +459,16 @@ def get_profile(request):
     if is_authed and not profile:
         profile = sync_profile(request.user.username, request.user, hide_profile=False)
 
+    if profile is not None and (profile.chat_id is '' or profile.gitcoin_chat_access_token is ''):
+
+        try:
+            from chat.tasks import associate_chat_to_profile
+            created, profile = associate_chat_to_profile(profile)
```

Comment:
shouldnt this be `associate_chat_to_profile.delay(profile)`? its gonna increase the loadtime every time we call get_profile (which is everywhere)

Reply:
It should only do it if you don't have a chat_id or a chat_access_token yet, and its not a delayed function because we want to return the profile back to the scope updated.

Reply:
hmm for some reason when i run this on my profile in prod i get this

> HTTPError: 500 Server Error: Internal Server Error for url: https://chat.gitcoin.co:443/api/v4/users/mqe31q5oo3bctd9qzqgtdco1kw/tokens

so as it stands this will run every time i run `get_profile()` as i never successfully get a chat access token